File: android/ac-teams/src/main/kotlin/com/microsoft/adaptivecards/teams/TeamsFluentTheme.kt
package com.microsoft.adaptivecards.teams

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color

object TeamsFluentTheme {
    @Composable
    fun getColorScheme(teamsTheme: TeamsTheme): ColorScheme {
        return when (teamsTheme) {
            TeamsTheme.LIGHT -> lightColorScheme(
                primary = Color(0xFF6264A7),
                background = Color.White,
                surface = Color(0xFFFAFAFA),
                onPrimary = Color.White,
                onBackground = Color.Black,
                onSurface = Color.Black
            )
            TeamsTheme.DARK -> darkColorScheme(
                primary = Color(0xFF8587D3),
                background = Color(0xFF1F1F1F),
                surface = Color(0xFF2D2D2D),
                onPrimary = Color.White,
                onBackground = Color.White,
                onSurface = Color.White
            )
            TeamsTheme.HIGH_CONTRAST -> darkColorScheme(
                primary = Color.Yellow,
                background = Color.Black,
                surface = Color(0xFF0D0D0D),
                onPrimary = Color.Black,
                onBackground = Color.White,
                onSurface = Color.White
            )
        }
    }
}

File: ios/Sources/ACTeams/TeamsFluentTheme.swift
import SwiftUI

public struct TeamsFluentTheme {
    public let colorScheme: ColorScheme
    public let primaryColor: Color
    public let backgroundColor: Color
    public let surfaceColor: Color
    public let textColor: Color
    
    public init(teamsTheme: TeamsTheme) {
        switch teamsTheme {
        case .light:
            self.colorScheme = .light
            self.primaryColor = Color(red: 0.38, green: 0.47, blue: 0.91)
            self.backgroundColor = .white
            self.surfaceColor = Color(white: 0.98)
            self.textColor = .black
        case .dark:
            self.colorScheme = .dark
            self.primaryColor = Color(red: 0.52, green: 0.60, blue: 0.95)
            self.backgroundColor = Color(white: 0.12)
            self.surfaceColor = Color(white: 0.18)
            self.textColor = .white
        case .highContrast:
            self.colorScheme = .dark
            self.primaryColor = .yellow
            self.backgroundColor = .black
            self.surfaceColor = Color(white: 0.05)
            self.textColor = .white
        }
    }
    
    public func apply() -> some View {
        Color.clear
            .preferredColorScheme(colorScheme)
    }
}
